### DISCUSSION TITLE
Custom emails on password reset and sign up

### DISCUSSION THREAD

**Discussion by abjelosevic88** [supabase/supabase#1666] [Questions] [answered]
Hi,

Is it possible to send emails manually using some custom SMTP provider like Mailjet or Mandrill on user sing up and password recovery?

If it is, how can I generate password recovery link manually?

Thanks

**Comment by awalias** [comment, marked as answer]
Hey @abjelosevic88 yes you can plug in a custom provider in the dashboard under > Auth > Settings > Configuration:

![image](https://user-images.githubusercontent.com/458736/121112833-98643480-c843-11eb-9cb1-2833587174c2.png)

then you can customize the email templates in Auth > Templates (they will still be used even for custom SMTP servers)

![image](https://user-images.githubusercontent.com/458736/121112899-b762c680-c843-11eb-8d22-7422df794813.png)

if you still want the password recovery link delivered programmatically we have a couple of issues open for this:
https://github.com/supabase/gotrue/issues/88
https://github.com/supabase/gotrue/issues/65 

hope this helps!

**Reply by abjelosevic88** [reply]
Wow, somehow I overlooked this. This is exactly what I was looking for! Thank you for your work and this amazing product!
